packages/ui/__tests__/components/molecules/RightIconButton.test.js

```diff
@@ -0,0 +1,54 @@
+/*
+ * Axelor Business Solutions
+ *
+ * Copyright (C) 2023 Axelor (<http://axelor.com>).
+ *
+ * This program is free software: you can redistribute it and/or  modify
+ * it under the terms of the GNU Affero General Public License, version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React from 'react';
+import {TouchableOpacity} from 'react-native';
+import {shallow} from 'enzyme';
+import {RightIconButton, Text, Icon} from '@axelor/aos-mobile-ui';
+
+describe('RightIconButton Component', () => {
+  const props = {
+    onPress: jest.fn(),
+    title: 'Click Me',
+    icon: <Icon name="right-icon" />,
+  };
+
+  it('should render without crashing', () => {
+    const wrapper = shallow(<RightIconButton {...props} />);
+    expect(wrapper.exists()).toBe(true);
+  });
+
+  it('displays the title when provided', () => {
+    const wrapper = shallow(<RightIconButton {...props} />);
+
+    expect(wrapper.find(Text).prop('children')).toBe(props.title);
+  });
+
+  it('displays the icon', () => {
+    const wrapper = shallow(<RightIconButton {...props} />);
+
+    expect(wrapper.contains(props.icon)).toBe(true);
+  });
+
+  it('calls onPress when the button is pressed', () => {
+    const wrapper = shallow(<RightIconButton {...props} />);
+    const touchable = wrapper.find(TouchableOpacity);
+    touchable.simulate('press');
+    expect(props.onPress).toHaveBeenCalled();
```

Comment:
This test will always return true even if TouchableOpacity is disabled (because of ReactNative) so when you don't need to check args provided, please only check that TouchableOpacity is not disabled.